pkg/tests/ossm/prometheus_scoped_scraping_test.go

```diff
@@ -0,0 +1,260 @@
+package ossm
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/maistra/maistra-test-tool/pkg/app"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/common"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/require"
+	"github.com/maistra/maistra-test-tool/pkg/util/curl"
+	namespaces "github.com/maistra/maistra-test-tool/pkg/util/ns"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+func TestOperatorCanUpdatePrometheusConfigMap(t *testing.T) {
+	test.NewTest(t).Groups(test.Full).Run(func(t test.TestHelper) {
+		t.Log("This test checks if the operator can update Prometheus ConfigMap when the SMMR is updated")
+
+		t.Cleanup(func() {
+			oc.ApplyString(t, meshNamespace, smmr)
+		})
+
+		prometheusPodSelector := pod.MatchingSelector("app=prometheus,maistra-control-plane=istio-system", meshNamespace)
+
+		checkPermissionError := func(t test.TestHelper) {
```

Comment:
Any reason why this is an anonymous instead of a regular function?

Reply:
Both `checkPermissionError` and `checkConfigurationReloadingTriggered` are using the `prometheusPodSelector` variable. This is why I choose them to be anonymous function.

Reply:
Ok, but you could also move the pod selector variable outside of the test function (or simply duplicate it). 

My thinking is that when one looks at a test function, they should be able to quickly tell what the test is doing (i.e. the code in the test function should be at a relatively high abstraction level, with the low-level stuff hidden behind functions). But these anonymous functions are fairly low level and they appear at the start of the test function, forcing you to read them first and causing you to learn about the test from the bottom up instead of top to bottom. 

Just my two cents... I'm fine with keeping these functions here.